bagitobjecttransfer/recordtransfer/forms/transfer_forms.py

```diff
@@ -902,4 +1006,137 @@ class ReviewForm(TransferForm):
     it.
     """
 
+    class Meta:
+        """Meta information for the form."""
+
+        transfer_step = TransferStep.REVIEW
+
     captcha = ReCaptchaField(widget=ReCaptchaV2Invisible, label="hidden")
+
+    @staticmethod
+    def _get_base_fields_data(form: BaseForm) -> dict[str, Any]:
+        """Extract fields data from a form."""
+        return {
+            form.fields[field].label or field: form.cleaned_data.get(field, "")
+            for field in form.fields
+            if form.fields[field].label != "hidden"
+        }
+
+    @staticmethod
+    def _process_file_upload(form: UploadFilesForm, user: User) -> dict[str, Any]:
+        """Handle file upload specific field processing."""
+        fields_data = ReviewForm._get_base_fields_data(form)
+        session_token = form.cleaned_data.get("session_token")
+
+        if not (session_token and user):
+            return fields_data
+
+        session = UploadSession.objects.filter(token=session_token, user=user).first()
+        if not session:
+            return fields_data
+
+        # Adds on links to access uploaded files
+        fields_data["Uploaded files"] = [
+            {"name": f.name, "url": f.get_file_access_url()}
+            for f in session.get_temporary_uploads()
+        ]
+
+        return fields_data
+
+    @staticmethod
+    def _process_group_transfer(form: GroupTransferForm, user: User) -> dict[str, Any]:
```

Comment:
To make consistent with `_get_base_fields_data`:

```suggestion
    def _get_group_transfer_fields_data(form: GroupTransferForm, user: User) -> dict[str, Any]:
```